File: app/src/main/java/de/wollis_page/gibsonos/activity/base/AppCompatListActivity.java
package de.wollis_page.gibsonos.activity.base;

import android.widget.ListAdapter;
import android.widget.ListView;

public abstract class AppCompatListActivity extends GibsonOsActivity {
    private ListView listView;

    protected ListView getListView() {
        if (listView == null) {
            listView = findViewById(android.R.id.list);
        }

        return listView;
    }

    protected void setListAdapter(ListAdapter adapter) {
        getListView().setAdapter(adapter);
    }
}

File: app/src/main/java/de/wollis_page/gibsonos/activity/base/AppCompatListActivity.kt
package de.wollis_page.gibsonos.activity.base

import android.widget.ListAdapter
import android.widget.ListView

abstract class AppCompatListActivity : GibsonOsActivity() {
    protected var listView: ListView? = null
        get() {
            if (field == null) {
                field = findViewById(android.R.id.list)
            }
            return field
        }
        private set

    protected fun setListAdapter(adapter: ListAdapter?) {
        listView!!.adapter = adapter
    }
}